jobs.py

```diff
@@ -332,29 +332,41 @@
     "tests/migration/test_host_evacuate.py::TestHostEvacuateWithNetwork",
 ]
 
-def get_vm_or_vms_refs(handle):
+def get_vm_or_vms_refs(handle, host_version=None):
     try:
-        from vm_data import VMS
+        from vm_data import get_vms
+        vms = get_vms(host_version)
     except ImportError:
         print("ERROR: Could not import VMS from vm_data.py.")
         print("Get the latest vm_data.py from XCP-ng's internal lab or copy data.py-dist and fill with your VM refs.")
         print("You may also bypass this error by providing your own --vm parameter(s).")
         sys.exit(1)
 
     category, key = handle.split("/")
-    if category not in VMS or not VMS[category].get(key):
+    if category not in vms or not vms[category].get(key):
         print(f"ERROR: Could not find VMS['{category}']['{key}'] in vm_data.py, or it's empty.")
         print("You need to update your local vm_data.py.")
         print("You may also bypass this error by providing your own --vm parameter(s).")
         sys.exit(1)
-    return VMS[category][key]
+    return vms[category][key]
 
 def build_pytest_cmd(job_data, hosts=None, pytest_args=[]):
     markers = job_data.get("markers", None)
     name_filter = job_data.get("name_filter", None)
 
     job_params = dict(job_data["params"])
 
+    host_version = None
+    if hosts is not None:
+        try:
+            host = hosts.split(',')[0]
+            cmd = ["ssh", host, "lsb_release", "-sr"]
+            res = subprocess.run(cmd, capture_output=True, text=True)
+            if res.returncode == 0:
+                host_version = res.stdout.strip()
+        except Exception as e:
```

Comment:
We assume all hosts provided have same versions, is it wanted?

Reply:
Gael and Sam will confirm but yes, all hosts are supposed to run the same XCP-ng version.